src/Api/AuthAction.php

```diff
@@ -19,27 +19,49 @@
 
 class AuthAction implements Action
 {
+    /**
+     * @var SettingsRepository
+     */
     protected $settings;
 
+    /**
+     * @var Pusher
+     */
+    protected $pusher;
+
     public function __construct(SettingsRepository $settings)
     {
         $this->settings = $settings;
+
+        $this->pusher = new Pusher(
+            $this->settings->get('pusher.app_key'),
+            $this->settings->get('pusher.app_secret'),
+            $this->settings->get('pusher.app_id')
+        );
```

Comment:
Can we merge this first and then move the code? That seems more "decent", as it's a different issue. I'll open up an issue that handles this.